typescript/packages/plugins/balmy/src/balmy.service.ts

```diff
@@ -0,0 +1,102 @@
+import { QuoteRequest, buildSDK } from "@balmy/sdk";
+import { Tool } from "@goat-sdk/core";
+import { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { ExecuteSwapParameters, GetQuoteParameters } from "./parameters";
+
+export class BalmyService {
+    private sdk;
+
+    constructor() {
+        this.sdk = buildSDK();
+    }
+
+    @Tool({
+        description: "Get quotes for a token swap using Balmy",
+    })
+    async getQuote(walletClient: EVMWalletClient, parameters: GetQuoteParameters) {
+        const chainid = await walletClient.getChain().id;
+
+        const request: QuoteRequest = {
+            chainId: chainid,
+            sellToken: parameters.tokenIn,
+            buyToken: parameters.tokenOut,
+            order:
+                parameters.order.type === "sell"
+                    ? {
+                          type: "sell",
+                          sellAmount: parameters.order.Amount,
+                      }
+                    : {
+                          type: "buy",
+                          buyAmount: parameters.order.Amount,
+                      },
+            slippagePercentage: parameters.slippagePercentage,
+            gasSpeed: parameters.gasSpeed,
+            takerAddress: parameters.takerAddress || "0x0000000000000000000000000000000000000000",
```

Comment:
Shouldn't the taker address default to the client wallet?

Reply:
I'd specify that in the parameter description too

Reply:
no the taker address is the one who will fill the order so def wont be the client , with 0x000 anyone can be that

Reply:
described